src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -1,31 +1,87 @@
 package mesosphere.marathon
 package api.akkahttp.v2
 
-import akka.http.scaladsl.model.StatusCodes
+import java.time.Clock
+
+import akka.event.EventStream
+import akka.http.scaladsl.model.{ Uri, StatusCodes }
+import akka.http.scaladsl.model.headers.Location
 import akka.http.scaladsl.server.Route
-import mesosphere.marathon.api.akkahttp.Controller
-import mesosphere.marathon.api.akkahttp.PathMatchers.PodsPathIdLike
+import mesosphere.marathon.api.akkahttp.{ Controller, Headers }
+import mesosphere.marathon.api.akkahttp.PathMatchers.{ PodsPathIdLike, forceParameter }
 import mesosphere.marathon.core.group.GroupManager
 import mesosphere.marathon.core.instance.Instance
-import mesosphere.marathon.plugin.auth.Authenticator
+import mesosphere.marathon.plugin.auth.{ Authenticator, Authorizer, CreateRunSpec }
 import mesosphere.marathon.state.PathId
 import akka.http.scaladsl.server.PathMatchers
+import mesosphere.marathon.api.v2.PodNormalization
+import mesosphere.marathon.api.v2.validation.PodsValidation
 import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.core.event.PodEvent
+import mesosphere.marathon.core.plugin.PluginManager
+import mesosphere.marathon.core.pod.PodManager
+import mesosphere.marathon.raml.{ PodConversion, Raml }
+
+import async.Async._
+import scala.concurrent.ExecutionContext
 
 class PodsController(
+    val config: MarathonConf,
     val electionService: ElectionService,
-    val groupManager: GroupManager)(
+    val podManager: PodManager,
+    val groupManager: GroupManager,
+    val pluginManager: PluginManager,
+    val eventBus: EventStream,
+    val clock: Clock)(
     implicit
-    val authenticator: Authenticator) extends Controller {
+    val authorizer: Authorizer,
+    val authenticator: Authenticator,
+    val executionContext: ExecutionContext) extends Controller {
 
   import mesosphere.marathon.api.akkahttp.Directives._
+  import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+
+  val podNormalizer = PodNormalization.apply(PodNormalization.Configuration(
+    config.defaultNetworkName.get))
 
   def capability(): Route =
     authenticated.apply { implicit identity =>
-      complete(StatusCodes.OK)
+      complete((StatusCodes.OK, ""))
     }
 
-  def create(): Route = ???
+  @SuppressWarnings(Array("all")) // async/await
+  def create(): Route =
+    authenticated.apply { implicit identity =>
+      (extractClientIP & forceParameter) { (clientIp, force) =>
+        extractRequest { req =>
+          entity(as[raml.Pod]) { podDef =>
+            val normalizedPodDef = podNormalizer.normalized(podDef)
```

Comment:
maybe I am missing something but how do we deal with NormalizationException here? Thanks

Reply:
Good point. I don't think they should be handled here but outside. What kind of normalization exceptions do we have? How can this be?

Reply:
How about having some kind of normalization directive? It will normalize and reject in case of errors.

Reply:
Yeah, I am not saying it should be handled here, I was just looking for a place where that is handled both here and in AppsController and did not find one... I am not against directive since that is how we handle validation errors too...

Reply:
Do you know cases where the `NormalizationException` is thrown? I could not find it for pods.

Reply:
Oh, the `PodNormalizationTest` have a case. However, it does not make much sense to me to have a normalization that can fail.

Reply:
well I found this and it is used in PodNormalization https://github.com/mesosphere/marathon/blob/77ccca15b91d86e71736961f6ba00155eb27e45f/src/main/scala/mesosphere/marathon/api/v2/NetworkNormalization.scala#L37

Reply:
Ok, I'll add a handler.

Reply:
It is so strange. #5741 introduces a test which should result in this kind of error. However, it is a validation error. I'm really not sure is there should be normalization issues. In any case, that should be part of a separated diff.

Reply:
Hm, it seems we do not handle `NormalizationException` at all.